hub/apps/desktop/modernize/xaml-islands.md

```diff
@@ -132,6 +132,8 @@ The following sections discuss limitations and workarounds for certain UWP devel
 
 :heavy_check_mark: Using `x:Bind` with hosted controls in XAML Islands is not supported. You'll have to declare the data model in a .NET Standard library.
 
+:heavy_check_mark: To support Narrator announcing text via [AutomationPeer.RaiseNotificationEvent](/uwp/api/windows.ui.xaml.automation.peers.automationpeer.raisenotificationevent) in a Win32 app, your XamlRoot's [Content](/uwp/api/windows.ui.xaml.xamlroot.content) must be able to create an automation peer or you need to set its [AutomationProperties.LandmarkTypeProperty](/uwp/api/windows.ui.xaml.automation.automationproperties.landmarktypeproperty) to [UIA_MainLandmarkTypeId](/windows/win32/winauto/landmark-type-identifiers). Also if your window is created using the `WS_POPUP` style you'll have to handle the [WM_GETOBJECT](/windows/win32/winauto/wm-getobject) message ([example](/win32/winauto/uiauto-howto-expose-serverside-uiautomation-provider)) and implement the [Window Control Pattern](/windows/win32/winauto/uiauto-implementingwindow).
```

Comment:
this behavior isn't specific to xaml islands. This caveat should go into the UiaRaiseNotificationEvent API: https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcoreapi/nf-uiautomationcoreapi-uiaraisenotificationevent

Reply:
OK, I'll make a PR for that page. But how would someone running into the issue we had when using AutomationPeer.RaiseNotificationEvent in XAML Islands with `WS_POPUP` know to look at the docs for UiaRaiseNotificationEvent? Does AutomationPeer.RaiseNotificationEvent always call UiaRaiseNotificationEvent or is that only in Islands win32 apps? Would it make sense to add a note to AutomationPeer.RaiseNotificationEvent docs to mention that in Islands it forwards to UiaRaiseNotificationEvent?

Reply:
> Does AutomationPeer.RaiseNotificationEvent always call UiaRaiseNotificationEvent or is that only in Islands win32 apps?

It will always call into UIA since that's the only way to interact with narrator (not counting things like the legacy automation frameworks which xaml doesn't use, like MSAA or oleacc).

> Does AutomationPeer.RaiseNotificationEvent always call UiaRaiseNotificationEvent or is that only in Islands win32 apps? Would it make sense to add a note to AutomationPeer.RaiseNotificationEvent docs to mention that in Islands it forwards to UiaRaiseNotificationEvent?

If you look at the [docs for AutomationPeer](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer?view=winrt-20348#remarks), you can see it says:

> AutomationPeer is the class that provides almost all of the API that eventually forwards the Microsoft UI Automation information for a UWP app using C++, C#, or Visual Basic to a Microsoft UI Automation client. Typical Microsoft UI Automation clients aren't calling AutomationPeer methods directly. These clients are often assistive technology that are using other programming models and operating as services, and are not likely to be calling Windows Runtime methods. But the general Microsoft UI Automation support in the Windows Runtime forwards all this information using the provider side of the Microsoft UI Automation framework. Any clients to Microsoft UI Automation can interact with the representative automation trees of a UWP app using C++, C#, or Visual Basic.

Reply:
New PR: https://github.com/MicrosoftDocs/sdk-api/pull/905